app/server.js

```diff
@@ -0,0 +1,276 @@
+    res.send(`user ${user.login} acquired package ${formDetails.package_id}`)    
+  }
+  catch (e) {
+    res.send(e);
+    printMessage('got error', e);
+  }
+
+}));
+
+app.get('/repositories', ensureAuthenticated, (req, res) => {
+
+  const user = req.session.user;
+  printMessage(user);
+
+  const storeRepositories = (body) => {
+    let repositories = JSON.parse(body);
+
+    coForEach(repositories, function* (item, index) {
+      let queryString = `INSERT INTO repositories (user_id, git_project_id, project_name, full_project_name, html_url, description, api_url)
+        SELECT '${user.id}', '${item.id}','${item.name}', '${item.full_name}', '${item.html_url}', '${item.description}', '${item.url}'
+        ON CONFLICT (git_project_id) DO UPDATE
+        SET project_name='${item.name}', full_project_name='${item.full_name}', html_url='${item.html_url}', description='${item.description}', api_url='${item.url}';`
+
+      yield dbService.queryDatabase(queryString);
+
+    })
+    res.render('repositories', { repositories: repositories });
+  }
+
+  try {
+    requestClient.get(`https://api.github.com/users/${user.login}/repos`)
+      .then(storeRepositories)
+  } catch (e) {
+    printMessage('got error', e);
+  }
+
+});
+
+
+app.get(`/repositories/issues/:name`, ensureAuthenticated, (req, res) => {
+
+  const user = req.session.user;
+
+  const storeIssues = (body) => { 
+    let issues = JSON.parse(body); 
+    printMessage(issues);
+
+    coForEach(issues, function* (item, index) {
+      let queryString = `INSERT INTO issues (url, repository_url, git_issue_id, title, user_id, body)
+        SELECT '${item.url}', '${item.repository_url}', ${item.id}, '${item.title}', ${user.id}, '${item.body}'
+        ON CONFLICT (git_issue_id) DO UPDATE
+        SET url='${item.url}', repository_url='${item.repository_url}', git_issue_id='${item.id}', title='${item.title}', user_id='${user.id}', body='${item.body}'`;
+
+      yield dbService.queryDatabase(queryString);
+
+    })
+    res.render('issues', { issues: issues })
+  }
+
+  try {
+    requestClient.get(`https://api.github.com/repos/${user.login}/${req.params.name}/issues`)
+      .then(storeIssues);
+
+  } catch (e) {
+    printMessage('got error', e);
+  }
+
+});
+
+app.get(`/packages`, ensureAuthenticated, co.wrap(function*(req, res) { 
+  
+  try {
+    let packages = [];
+    let queryString = `SELECT * FROM packages;`
+    yield dbService.queryDatabase(queryString)
+      .then((array) => array.forEach((item) => packages.push(item)));
+
+    printMessage('packages', packages.length);
+    res.render('packages', {packages: packages});
+  }
+  catch(e) {
+    res.send(e);
+    printMessage('got error', e);
+  }
+  
+}));
+
+app.post(`/create-package`, ensureAuthenticated, co.wrap(function*(req, res){
+
+  let formDetails = req.body;
+  let queryString = `INSERT INTO public.packages (name, price) VALUES('${formDetails.name}', ${formDetails.price});`;
+
+  try {
```

Comment:
zamiast try catchy możesz wywoływać funkcjonalności w chainie promisów. Pod tym względem są fajne, ponieważ przechwytują errory i następnie wystarczy, że wywołasz na końcu chaina .catch() i będziesz miał error handler w jednym miejscu. Jeśli dodatkowo zrobisz expressowy errorHandler to będziesz mógł mieć error handler w jednym miejscu, poniewać w każdym kontrolerze (który zadeklarujesz jako middleware) będziesz mógł wywołać na końcu chaina promisów .catch(next), co sprawi, że error trafi do globalnego errorHandlera (Sprawdź jak to wygląda w dokumentacji expressa)